pandas/_libs/interval.pyi

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations
+
+from typing import (
+    Any,
+    Generic,
+    Literal,
+    Tuple,
+    TypeVar,
+    Union,
+    overload,
+)
+
+import numpy as np
+import numpy.typing as npt
+
+from pandas._typing import (
+    Timedelta,
+    Timestamp,
+)
+
+VALID_CLOSED: frozenset[str]
+
+_OrderableScalarT = TypeVar("_OrderableScalarT", int, float)
+_OrderableTimesT = TypeVar("_OrderableTimesT", Timestamp, Timedelta)
+_OrderableT = TypeVar("_OrderableT", int, float, Timestamp, Timedelta)
+
+class _LengthDescriptor:
+    @overload
+    def __get__(self, instance: Interval[float], owner: Any) -> float: ...
+    @overload
+    def __get__(self, instance: Interval[int], owner: Any) -> int: ...
```

Comment:
I think these two overloads could be:

`def __get__(self, instance: Interval[_OrderableScalarT], owner: Any) -> _OrderableScalarT:`

Reply:
fixed in next commit